.editorconfig

```diff
@@ -3,9 +3,55 @@
 # top-most EditorConfig file
 root = true
 
-[*.json]
+[*]
 indent_style = space
 indent_size = 2
+end_of_line = lf
```

Comment:
With this, if you're using vscode, when you save a file it will be saved with `lf` and if you're using vscode to commit it will automatically ignore line ending changes.

Reply:
I know what you are saying, and you are correct that git can manage line endings. The default behavior is git will automatically decide what line endings to push. The only time I've seen this be an issue is when using wsl it will commit crlf if they exist, which shouldn't happen if editor config exists and vs code has saved the file.

That being said I also support using git attributes and can add that to this PR tomorrow.

Reply:
As long as you have `* text=auto`, git will never commit `CRLF`. Line endings in the repo (what you commit) will always be `LF`. However, what you get when you *check out files* can either be `LF` or `CRLF` depending on settings in the `.gitattributes`.

The EOL setting in the `.editorconfig` won't affect what line endings get committed to the repo itself, but it does affect what line endings the files have after checkout. On Windows, Git will check out files with `CRLF` line endings unless you specify `eol=lf` for file patterns in `.gitattributes`. You can set up your `.editorconfig` to use LF line endings when it saves files. Git does issue warnings when line endings do not match the expected format; in this case, you'll get a warning that LF line endings will be replaced with CRLF on checkout (when git overwrites the file).

There's a lot more to it, but that's the gist. There's several git configuration settings that also impact this behavior, but those should be avoided (and `.gitattributes` takes precedence anyway for most of them).